ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java

```diff
@@ -107,13 +110,21 @@ public void stop(final QueryId queryId) {
     log.info("Attempting to stop Query: " + queryId.toString());
     if (collocatedQueries.containsKey(queryId) && sources.containsKey(queryId)) {
       if (kafkaStreams.state().isRunningOrRebalancing()) {
-        kafkaStreams.removeNamedTopology(queryId.toString());
+        try {
+          kafkaStreams.removeNamedTopology(queryId.toString(), true)
+              .all()
+              .get(30, TimeUnit.SECONDS);
+          kafkaStreams.cleanUpNamedTopology(queryId.toString());
+        } catch (final TimeoutException | ExecutionException | InterruptedException e) {
+          e.printStackTrace();
+          throw new IllegalStateException(
```

Comment:
Are we sure this should be an IllegalStateException? What do we throw (if anything) if we fail to close a regular/non-binpacked query in the allotted timeout (ie if `KafkaStreams.close(timeout)` returns false)?

Reply:
We just warn. I am not sure that is the right thing to do though. But that can work for now